### PR TITLE
fix #2467 Get rid of Hamcrest matchers

### LINKED ISSUE #2467: Remove last traces of JUnitParams and Hamcrest in 3.3+
After #2442, we don't have JUnit4 matchers but we still have Hamcrest matchers in a couple of places.
Note that it won't really be possible to entirely remove the transitive dependency to hamcrest, but we can replace leftover usages with AssertJ... About 28 usages in `FluxTests.java` and 19 in `MonoTests.java`.

Also, it seems that junitparams dependency was removed, but the `jUnitParamsVersion` variable is still there in `build.gradle` and can be removed.

### PR DESCRIPTION
**Pull request by ericbottard**
Also remove a now unused gradle version property.

### CODE REVIEW

File: reactor-core/src/test/java/reactor/core/publisher/scenarios/FluxTests.java [hunk 1]
```diff
@@ -1136,7 +1127,7 @@ public void consistentMultithreadingWithPartition() throws InterruptedException
 
 
 		latch.await(30, TimeUnit.SECONDS);
-		assertThat("Not totally dispatched: " + latch.getCount(), latch.getCount() == 0);
+		assertThat(latch.getCount()).as("Not totally dispatched: ").isEqualTo(0L);
```
Comment (bsideup) [inline reactor-core/src/test/java/reactor/core/publisher/scenarios/FluxTests.java:1130]: unlike with `assertThat(String, boolean)` where String is the explanation message, `as` accepts the description of the value, so it shouldn't be inverted

**Review by simonbasle — APPROVED** [review]

**Review by bsideup — APPROVED** [review]

**Review by simonbasle — CHANGES_REQUESTED** [review]
I think that comment applies to more or less all the occurrences of `as` that are changed in this PR. it was also the case in the Junit4->AssertJ migration, but here the scope is much more narrow and manageable. what I generally recommend is that `as` reflects purely the variable being asserted, not the assertion itself or error message.

**Comment by ericbottard** [comment]
Went ahead and rephrased all `.as()` of this class. As Simon says, there could be value in a big sweep, but at least did this class.

**Comment by simonbasle** [comment]
yeah to be clearer I was just advocating for covering files in this PR, since I told you not bother with that during the AssertJ PR review 👍

**Review by simonbasle — APPROVED** [review]

**Pull request merged**